### PR TITLE
fix(ProfileParameterValidator): Resolve NamedModels first

### PR DESCRIPTION
**Pull request by lukas-valenta**
## Description

Changes ProfileParameterValidator behavior to resolve NamedModels before field references.

## Motivation and Context

When the field references were resolved before named models, the model definitions were not found, thus crashing the validator.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/internal/interpreter/profile-parameter-validator.ts [hunk 1]
```diff
@@ -483,12 +483,12 @@ export class ProfileParameterValidator implements ProfileVisitor {
     if (!this.namedDefinitionsInitialized) {
       node.definitions
         .filter(
-          (definition): definition is NamedFieldDefinitionNode =>
-            definition.kind === 'NamedFieldDefinition'
+          (definition): definition is NamedModelDefinitionNode =>
+            definition.kind === 'NamedModelDefinition'
         )
```
Comment (martinalbert) [inline src/internal/interpreter/profile-parameter-validator.ts:488]: Why not use profile ast utils? `isNamedModelDefinitionNode` and `isNamedFieldDefinitionNode` should work.

Comment (martinalbert) [inline src/internal/interpreter/profile-parameter-validator.ts:488]: Why not use profile ast utils? `isNamedModelDefinitionNode` and `isNamedFieldDefinitionNode` should work.

**Review by martinalbert — COMMENTED** [review]
I had similar problem resolving fields and models in profile-io-analyzer so I tested whether there won't be problem when I define field first and use it in model and it passes, but based on `visitFieldDefinitionNode()`, it checks whether [field is initialized in](https://github.com/superfaceai/one-sdk-js/blob/64c8b09eedc3c8faa28775c28bec70f46d2152d2/src/internal/interpreter/profile-parameter-validator.ts#L252-L256) `this.namedFieldDefinitions`, which will be always undefined when models are initialized first. 🤷🏻‍♂️

File: src/internal/interpreter/profile-parameter-validator.ts (hunk 1 above)
Comment (lukas-valenta) [reply src/internal/interpreter/profile-parameter-validator.ts:488]: i just moved the code, but good point :+1:

**Comment by lukas-valenta** [comment]
i see, so it actually passes only because it's basically untyped :thinking: i'll think about it more and add some more tests

**Comment by lukas-valenta** [comment]
@martinalbert I tried it out and it actually resolves correctly, because the values are validated lazily. Please try and find a counterexample if you think there's still a problem. In ProfileIOAnalyzer it's not resolved lazily, so there's a problem.

**Review by martinalbert — APPROVED** [review]
Yeah, I tried to debug it and you're right. 👍🏻

**Pull request merged**